File: Program's_Contributed_By_Contributors/C++ Programs/factorial.cpp
#include <stdio.h>

// my factorial -> nx(n-1)x(n-2)x(n-3)x...
int factorial(int number) {
  if (number==0) return 1;
  else return (number * factorial(number-1));
}

int main() {

  int number;
  scanf("%d", &number);
  printf("The factorial of %d : %d\n", number, factorial(number));
  return 0;
}

File: Factorial.java
// Java program to find factorial of given number
class Test
{
// method to find factorial of given number
static int factorial(int n)
{
if (n == 0)
return 1;

return n*factorial(n-1);
}

// Driver method
public static void main(String[] args)
{
int num = 5;
System.out.println("Factorial of "+ num + " is " + factorial(5));
}
}
